### ISSUE TITLE
Device is not discovered

### ISSUE THREAD

**Issue by MarvinSauthoff** [Aircoookie/Espalexa#193]
I used the example sketch "EspalexaColor" and added "#define ESPALEXA_DEBUG" before "#include <Espalexa.h>" to see debug output.
It seems that espalexa didn't recognize the discovery of my echo because while my echo is searching for devices espalexa didn't output anything...

**Comment by robertwillian** [comment]
Try this:
https://github.com/Aircoookie/Espalexa/pull/194

**Comment by MarvinSauthoff** [comment]
Thanks for your replay but unfortunately, this did not bring any improvement.
I dont't think that this is any kind of crash because in serial monitor i can't see anything which points on that.
If Espalexa is not writing any debug output while my echo is discovering new devices...

**Comment by FrankHovis** [comment]
I'm getting this output in Debug mode using an ESP32

Constructing device 1
Adding device 1
Constructing device 2
Adding device 2
Adding device 3
Espalexa Begin...
MAXDEVICES 10
Done
[MAIN] Free heap: 72256 bytes
Got UDP!
M-SEARCH * HTTP/1.1
HOST: 239.255.255.250:1900
ST: ssdp:all
MAN: "ssdp:discover"
MX: 3


Responding search req...
Got UDP!
M-SEARCH * HTTP/1.1
HOST: 239.255.255.250:1900
ST: upnp:rootdevice
MAN: "ssdp:discover"
MX: 3


Responding search req...
Got UDP!
M-SEARCH * HTTP/1.1
HOST: 239.255.255.250:1900
ST: ssdp:all
MAN: "ssdp:discover"
MX: 3


Responding search req...
Got UDP!
M-SEARCH * HTTP/1.1
HOST: 239.255.255.250:1900
ST: upnp:rootdevice
MAN: "ssdp:discover"
MX: 3


Responding search req...
Got UDP!
M-SEARCH * HTTP/1.1
HOST: 239.255.255.250:1900
ST: ssdp:all
MAN: "ssdp:discover"
MX: 3


Responding search req...
Got UDP!
M-SEARCH * HTTP/1.1
HOST: 239.255.255.250:1900
ST: upnp:rootdevice
MAN: "ssdp:discover"
MX: 3


Responding search req...
# Responding to description.xml ... #

Send setup.xml
<?xml version="1.0" ?><root xmlns="urn:schemas-upnp-org:device-1-0"><specVersion><major>1</major><minor>0</minor></specVersion><URLBase>http://192.168.0.15:80/</URLBase><device><deviceType>urn:schemas-upnp-org:device:Basic:1</deviceType><friendlyName>Espalexa (192.168.0.15:80)</friendlyName><manufacturer>Royal Philips Electronics</manufacturer><manufacturerURL>http://www.philips.com</manufacturerURL><modelDescription>Philips hue Personal Wireless Lighting</modelDescription><modelName>Philips hue bridge 2012</modelName><modelNumber>929000226503</modelNumber><modelURL>http://www.meethue.com</modelURL><serialNumber>94b97ec4a784</serialNumber><UDN>uuid:2f402f80-da50-11e1-9b23-94b97ec4a784</UDN><presentationURL>index.html</presentationURL></device></root>
Not-Found HTTP call:
URI: /api
Body: {"devicetype": "Echo"}
AlexaApiCall
ok
devType
Not-Found HTTP call:
URI: /api/2WLEDHardQrI3WHYTHoMcXHgEspsM8ZZRpSKtBQr/lights
Body: 
AlexaApiCall
ok
l0
lAll
Not-Found HTTP call:
URI: /api
Body: {"devicetype": "Echo"}
AlexaApiCall
ok
devType
Not-Found HTTP call:
URI: /api/2WLEDHardQrI3WHYTHoMcXHgEspsM8ZZRpSKtBQr/lights
Body: 
AlexaApiCall
ok
l0
lAll
Not-Found HTTP call:
URI: /api/2WLEDHardQrI3WHYTHoMcXHgEspsM8ZZRpSKtBQr/lights/1649656320
Body: 
AlexaApiCall
ok
l1649656320
1649656320
Not-Found HTTP call:
URI: /api/2WLEDHardQrI3WHYTHoMcXHgEspsM8ZZRpSKtBQr/lights/1649656321
Body: 
AlexaApiCall
ok
l1649656321
1649656321
Not-Found HTTP call:
URI: /api/2WLEDHardQrI3WHYTHoMcXHgEspsM8ZZRpSKtBQr/lights/1649656322
Body: 
AlexaApiCall
ok
l1649656322
1649656322
[MAIN] Free heap: 69524 bytes
Not-Found HTTP call:
URI: /api/2WLEDHardQrI3WHYTHoMcXHgEspsM8ZZRpSKtBQr/lights/1649656320
Body: 
AlexaApiCall
ok
l1649656320
1649656320
Not-Found HTTP call:
URI: /api/2WLEDHardQrI3WHYTHoMcXHgEspsM8ZZRpSKtBQr/lights/1649656321
Body: 
AlexaApiCall
ok
l1649656321
1649656321
Not-Found HTTP call:
URI: /api/2WLEDHardQrI3WHYTHoMcXHgEspsM8ZZRpSKtBQr/lights/1649656322
Body: 
AlexaApiCall
ok
l1649656322
1649656322
[MAIN] Free heap: 68884 bytes
[MAIN] Free heap: 68884 bytes
[MAIN] Free heap: 68884 bytes

Alexa isnt finding anything

**Comment by amitesh-singh** [comment]
I am also facing same issue. 
did  you find the fix?

**Comment by MarvinSauthoff** [comment]
No, unfortunately not.Am 08.01.22, 16:30 schrieb Amitesh Singh ***@***.***>:

   
  I am also facing same issue. did you find the fix?

**Comment by enricoperfetto** [comment]
Same issue for me. I am using ESPALEXA_ASYNC and library version 2.7.0
